dpnp/backend/kernels/dpnp_krnl_indexing.cpp

```diff
@@ -26,12 +26,46 @@
 #include <iostream>
 #include <list>
 #include <vector>
+#include <list>
 
 #include <dpnp_iface.hpp>
 #include "dpnp_fptr.hpp"
 #include "dpnpc_memory_adapter.hpp"
 #include "queue_sycl.hpp"
 
+template <typename _DataType1, typename _DataType2>
+class dpnp_choose_c_kernel;
+
+template <typename _DataType1, typename _DataType2>
+void dpnp_choose_c(void* result1, void* array1_in, std::vector<void*> choices1, size_t size)
+{
+    if ((array1_in == nullptr) || (result1 == nullptr) || !choices1.size() || !size)
+    {
+        return;
+    }
+    DPNPC_ptr_adapter<_DataType1> input1_ptr(array1_in, size);
+    _DataType1* array_in = input1_ptr.get_ptr();
+
+    DPNPC_ptr_adapter<_DataType2*> choices_ptr(choices1.data(), choices1.size());
```

Comment:
Wrong usage.
In this code, potentially, the container itself could be copied. The main idea is to copy input memory from `device` to the kernel space. So, it looks like a loop needed here over input data stored in the container